backend/src/services/logbooks-service.js

```diff
@@ -1,232 +1,58 @@
+import getLogbookType from "../utils/get-logbook-type.js";
+import getTable from "../utils/get-table.js";
+import insertTable from "../utils/insert-table.js";
+import parseUserID from "../utils/parse-user-id.js";
+
 export async function createLogbook(req) {
-    try {
-        const supabase = req.supabase;
-        const { type } = req.body;
-        const { data, error } = await supabase.from("logbooks").insert({ type: type }).select();
-        if (error) {
-            throw new Error(error.message);
-        }
-        return data[0];
-    } catch (error) {
-        throw new Error(error.message);
-    }
+    const supabase = req.supabase;
+    const body = req.body
+    return insertTable(supabase, "logbooks", body);
 }
 
 export async function getUserLogbooks(req) {
-    try {
-        const supabase = req.supabase;
-        const token = req.header("Authorization")?.split(" ")[1];
-        const userID = parseUserID(token);
-        const { data, error } = await supabase.from("logbooks").select().eq("user_id", userID);
-        if (error) {
-            throw new Error(error.message);
-        }
-        return data;
-    } catch (error) {
-        throw new Error(error.message);
-    }
-}
-
-function parseUserID(token) {
-    const parts = token.split(".");
-    const decodedPayload = JSON.parse(atob(parts[1]));
-    const userID = decodedPayload["sub"];
-    return userID;
+    const supabase = req.supabase;
+    const token = req.header("Authorization")?.split(" ")[1];
+    const userID = parseUserID(token);
+    return getTable(supabase, "logbooks", "user_id", userID, "collection");
 }
 
 export async function getLogbook(req) {
```

Comment:
edge case here, if the logbook id is misspelled incorrect, the server will still crash... I get the following error message when i try this:
![image](https://github.com/user-attachments/assets/370c5d37-d8c3-4c6a-90c7-7f4d83e7080b)

Could you please look into this?